File: csharp/VS2008/netframework/Modules/10.API/30.FlexCelImageExplorer/TCompressForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.IO;

namespace FlexCelImageExplorer
{
    /// <summary>
    /// Summary description for TCompressForm.
    /// </summary>
    public partial class TCompressForm: System.Windows.Forms.Form
    {

        public TCompressForm()
        {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();
            cbPixelFormat.SelectedIndex = 2;

            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }

        private byte[] FImageToUse;
        private string FXlsFilename;

        internal byte[] ImageToUse
        {
            get
            {
                return FImageToUse;
            }
            set
            {
                FImageToUse = value;
                using (MemoryStream ms = new MemoryStream(value))
                {
                    pictureBox1.Image = Image.FromStream(ms);
                }
            }
        }

        internal string XlsFilename
        {
            get
            {
                return FXlsFilename;
            }
            set
            {
                FXlsFilename = value;
            }
        }

        private void btnCancel_Click(object sender, System.EventArgs e)
        {
            Close();
        }

        private void TCompressForm_Load(object sender, System.EventArgs e)
        {
        }

        private void btnOk_Click(object sender, System.EventArgs e)
        {
            Close();
        }
    }
}

File: vb/VS2017/netframework/Modules/10.API/30.FlexCelImageExplorer/TCompressForm.vb
Imports System.Collections
Imports System.ComponentModel
Imports System.IO

Namespace FlexCelImageExplorer
	''' <summary>
	''' Summary description for TCompressForm.
	''' </summary>
	Partial Public Class TCompressForm
		Inherits System.Windows.Forms.Form

		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()
			cbPixelFormat.SelectedIndex = 2

			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub

		Private FImageToUse() As Byte
		Private FXlsFilename As String

		Friend Property ImageToUse() As Byte()
			Get
				Return FImageToUse
			End Get
			Set(ByVal value As Byte())
				FImageToUse = value
				Using ms As New MemoryStream(value)
					pictureBox1.Image = Image.FromStream(ms)
				End Using
			End Set
		End Property

		Friend Property XlsFilename() As String
			Get
				Return FXlsFilename
			End Get
			Set(ByVal value As String)
				FXlsFilename = value
			End Set
		End Property

		Private Sub btnCancel_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnCancel.Click
			Close()
		End Sub

		Private Sub TCompressForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
		End Sub

		Private Sub btnOk_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnOk.Click
			Close()
		End Sub
	End Class
End Namespace
